src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -421,12 +459,21 @@
      * Query this inventory for inventories matching any of the supplied
      * queries. Logical <code>OR</code> is applied between operands.
      *
-     * @param <T> expected inventory type, specified as generic to allow easy
-     *      pseudo-duck-typing
      * @param operations queries to check against
      * @return the query result
      */
-    <T extends Inventory> T query(QueryOperation<?>... operations);
+    Inventory query(QueryOperation<?>... operations);
+
+    /**
+     * Query this inventory for a single inventory matching the supplied inventory type.
+     * This query will return {@link Optional#empty()} when the query does not return a
+     * single inventory matching the supplied inventory type.
+     *
+     * @param inventoryType The inventory type to query for
+     * @param <T> The Type of inventory
+     * @return the query result
+     */
+    <T extends Inventory> Optional<T> query(Class<T> inventoryType);
```

Comment:
Maybe rename this to `querySingle` (or something else)? The normal `query` method returns a collection `Inventory` and this method returning a single `Inventory` could make the naming confusing.

Reply:
@Faithcaio thoughts?